src/WebJobs.Script.WebHost/Diagnostics/WebHostMetricsLogger.cs

```diff
@@ -27,5 +29,10 @@ public void EndEvent(MetricEvent metricEvent)
                 MetricsEventManager.FunctionCompleted();
             }
         }
+
+        public void OnHostStartedEvent(ScriptHost instance)
```

Comment:
We want a general `void EndEvent(MetricEvent metricEvent)` method here. In WebJobs.Script.WebHost, add a new event class `HostStartedEvent : MetricEvent` that has the event data you need.  Then in WebHostMetricsLogger handle that event type.